### PR TITLE
Improve documentation with templates

### PR DESCRIPTION
**Pull request by MalloZup**
# what does this pr?

This pr add :

1) template documentation for our basic usages of variables

2) add documentation in this way for ISCSI usage.

# Description:

So basically for each useful scenario we consider important, we add a set of variables.

Imho we should end-up to have maximal 6/7 ( hana, hana monitoring, netweav., netweaver monitoring, isci , drbd etc).

This is a initial draft. I think we can improve it step by step later on, but at least it is a better then nothing.

I think we can live with the fact initally the templates are for libvirt. Later on we could add them for azure or whatever if needed. but the difference will be not big.

### CODE REVIEW

**Comment by Simranpal** [comment]
maybe we can create a template under directory for each provider

**Comment by MalloZup** [comment]
right now, I think we should just merge this PR since is imho an improvement. 

I would not imho provide to much nested doc with dir unless there is a customer saying it. 

The 1st use-case for this doc is already for us internally. we could indeed add azure later or other provider if we have needs. 

But right now, I would at least focus on libvirt where we need to provide netweaver doc and so on

File: README.md [hunk 1]
```diff
@@ -20,6 +20,14 @@ ___
 - [DRBD](doc/drbd.md)
 - [QA](doc/qa.md)
 - [Saptune](doc/saptune.md)
+- [ISCSI](doc/iscsi.md)
+
+## Templates:
+
+We provide minimal templates for setting up the minimal templates variables in order to get started with the project.
```
Comment (stefanotorresi) [inline README.md]: "minimal templates" is repeated.

```suggestion
We provide minimal templates for setting up the terraform variables in order to get started with the project.
```

File: doc/deployment-templates.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+# Deployement templates:
+
+This doc is an invetory of common functional templates, use-cases of the deployment project.
```
Comment (stefanotorresi) [inline doc/deployment-templates.md]: typo and wording

```suggestion
This document is an inventory of functional templates for common use-cases of the deployment project.
```

File: doc/deployment-templates.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+This examples were used with KVM libvrt but the other providers will have the same vars or differ not significantly.
```
Comment (stefanotorresi) [inline doc/deployment-templates.md]: typo and wording

```suggestion
The following examples refer to the libvirt Terraform provider, but they may be used for the other providers as well: there are no significant differences for the most common settings.
```

File: doc/deployment-templates.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+Additionally you need to set up pillars. In dev-mode we use mostly `automatic`. 
+See https://github.com/SUSE/ha-sap-terraform-deployments/tree/master/pillar_examples#pillar-examples for more details.
+
+
+# hana cluster
+
+```
+qemu_uri = "qemu+ssh://MYUSER@MYSTEM/system"
+base_image = "URL_TO_IMAGE"
+iprange = "192.168.210.0/24"
```
Comment (angelabriel) [inline doc/deployment-templates.md]: the iprange value (210) does not fit to your chosen host_ips (110) in line 26

in all of your examples

File: doc/deployment-templates.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+reg_code = "MY_REG_CODE"
+reg_email = "MY_EMAIL"
+ha_sap_deployment_repo = "http://download.opensuse.org/repositories/network:/ha-clustering:/Factory/SLE_15/"
+shared_storage_type = "shared-disk"
+storage_pool           = "terraform"
+```
+
+# monitoring hana cluster
+
+To monitoring ha_cluster you need only 2 vars more then hana_cluster deployment.
```
Comment (stefanotorresi) [inline doc/deployment-templates.md]: I don't think there is any need to use snake_case here.

```suggestion
To monitoring the HANA cluster need only 2 vars more than the simple HANA deployment.
```

File: doc/deployment-templates.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+# Deployement templates:
```
Comment (stefanotorresi) [inline doc/deployment-templates.md]: typo

```suggestion
# Deployment templates
```

File: README.md [hunk 7]
```diff
@@ -20,6 +20,14 @@ ___
 - [DRBD](doc/drbd.md)
 - [QA](doc/qa.md)
 - [Saptune](doc/saptune.md)
+- [ISCSI](doc/iscsi.md)
+
+## Templates:
```
Comment (stefanotorresi) [inline README.md]: I don't think we need colons in titles

```suggestion
## Templates
```

File: doc/deployment-templates.md (hunk 4 above)
Comment (MalloZup) [reply doc/deployment-templates.md]: thx

Comment (MalloZup) [reply doc/deployment-templates.md]: adapted doc accordingly

**Comment by MalloZup** [comment]
thx @stefanotorresi recheck. If ok merge tia

File: doc/deployment-templates.md (hunk 2 above)
Comment (stefanotorresi) [reply doc/deployment-templates.md]: I think you missed this one!

File: doc/deployment-templates.md (hunk 6 above)
Comment (stefanotorresi) [reply doc/deployment-templates.md]: and this one too

**Comment by stefanotorresi** [comment]
@MalloZup you missed a couple of typos suggested above

**Comment by MalloZup** [comment]
thx done

**Pull request merged**